app/domain/authentication/authn_gcp/restrictions_validator.rb

```diff
@@ -0,0 +1,105 @@
+module Authentication
+  module AuthnGCP
+
+    # This class is responsible of validating resource restrictions combination.
+    # the allowed combination is: Role should have at least 1 permitted constraints, and non of illegal constraints
+    class RestrictionsValidator
```

Comment:
@nessiLahav why is this class not a command class `ValidateResourceRestrictions`?

Reply:
so i'd like to take this example to learn from as a group. @nessiLahav says that he didn't implement this class as a command-class because it doesn't have any dependencies (apart from `logger` which is a somewhat special case). However, in the `README` that @jonahx wrote recently he separated the classes into 3 types:
- Command objects
- Value objects
- Objects with side effects

it is clear that this class is a Command Object as it performs some command (it validates resource restrictions). So from this, it seems that it should be a command class. However, i do remember a rule of thumb that if there are no dependencies then it shouldn't be a command class. 

So i'm quite confused. @jonahx care to jump in?